### PR TITLE
config: Move `discard_abci_responses` flag into its own storage section

### LINKED ISSUE #9274: config: Migrate `discard_abci_responses` from `[rpc]` to new `[storage]` section
As discussed in https://github.com/tendermint/tendermint/pull/9267#discussion_r946753672 and on Slack, we should migrate [this configuration option](https://github.com/tendermint/tendermint/blob/1f0cf7762bc414bf95b93afa773160f8467902e1/config/config.go#L412) to its own new section, as it seems a little misplaced to have a state store configuration option under the `[rpc]` section.

So far the best option seems to be a new `[storage]` section.

### PR DESCRIPTION
**Pull request by thanethomson**
Closes #9274 

Docs and `CHANGELOG_PENDING` entry will be added in #9267.

---

#### PR checklist

- [x] Tests written/updated, or no tests needed
- [x] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [x] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: config/toml.go [hunk 1]
```diff
@@ -483,6 +480,15 @@ create_empty_blocks_interval = "{{ .Consensus.CreateEmptyBlocksInterval }}"
 peer_gossip_sleep_duration = "{{ .Consensus.PeerGossipSleepDuration }}"
 peer_query_maj23_sleep_duration = "{{ .Consensus.PeerQueryMaj23SleepDuration }}"
 
+#######################################################
+###         Storage Configuration Options           ###
+#######################################################
+
+# Set to false to ensure ABCI responses are persisted. ABCI responses are
+# required for /block_results RPC queries, and to reindex events in the
+# command-line tool.
```
Comment (cmwaters) [inline config/toml.go:489]: Can we mention that setting this to true will save considerable disk space (or something similar)

**Review by cmwaters — APPROVED** [review]
Awesome. Thanks for doing that. Had half started last night but this is basically exactly how I had done it. I had even positioned this section specifically below the consensus config lol

File: config/toml.go (hunk 1 above)
Comment (thanethomson) [reply config/toml.go]: Does the new comment seem reasonable?

**Review by cmwaters — APPROVED** [review]
Yup the comments look good

**Pull request merged**